frontend/components/sidebar/Sidebar.tsx

```diff
@@ -88,14 +89,15 @@ const HideSidebarBtnContainer = styled.div`
   align-self: flex-end;
 `;
 
+// Props for the Sidebar component
 interface SideBarProps {
   selectedDb: AppState['selectedDb'];
   setSelectedDb: AppState['setSelectedDb'];
   setERView: AppState['setERView'];
   curDBType: DBType | undefined;
   setDBType: (dbType: DBType | undefined) => void;
   DBInfo: DatabaseInfo[] | undefined;
-  queryDispatch: ({ type, payload }) => void;
+  queryDispatch: ({ type, payload }: { type: string; payload?: any }) => void;
```

Comment:
Replace current Context API code with correct redux function calls. You will need to import those functions from the slices